hedera-node/hedera-app/src/test/java/com/hedera/node/app/blocks/BlockRetentionManagerTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2024-2025 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.blocks;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.hedera.node.app.blocks.impl.FileBlockItemWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class BlockRetentionManagerTest {
+    @TempDir
+    private Path uploadedDir;
+
+    private BlockRetentionManager blockRetentionManager;
+
+    private static final int AWAIT_SECONDS_TIMEOUT = 60;
+
+    @Test
+    void testCleanupShouldNotRunUntilScheduleTime() throws IOException {
+        final long retentionPeriodMs = 10;
+        final long cleanupPeriodMs = 100;
+        blockRetentionManager = new BlockRetentionManager(
+                uploadedDir, Duration.ofMillis(retentionPeriodMs), Duration.ofMillis(cleanupPeriodMs), 4);
+        final Runnable cleanupTask = () -> blockRetentionManager.startCleanup();
+        blockRetentionManager.scheduleRepeating(cleanupTask);
+
+        final var blockFileName = "file1" + "." + FileBlockItemWriter.RECORD_EXTENSION;
+
+        createTestFile(blockFileName);
+
+        await().atLeast(2 * retentionPeriodMs, TimeUnit.MILLISECONDS);
+        assertTrue(Files.exists(uploadedDir.resolve(blockFileName)), blockFileName + " should not have been deleted");
+
+        await().atMost(AWAIT_SECONDS_TIMEOUT, TimeUnit.SECONDS)
+                .until(() -> !Files.exists(uploadedDir.resolve(blockFileName)));
+        assertFalse(Files.exists(uploadedDir.resolve(blockFileName)), blockFileName + " should have been deleted");
```

Comment:
Hmm, up to 60 seconds is a looong time to wait for a unit test.. Not the end of the world, but it would improve our quality of life to figure out a way to make this test faster. I'll think about this 🤔

Reply:
In general we shouldn't be waiting as much and I set 60s just to make sure that if we fail here it will not be likely due to CI flakiness. I am ok lowering it to 5-10s instead (I see we use such values in other unit tests using `await()`), does that seem fine to you?

Reply:
Ideally we want dozens (or more) of unit tests to run in 5-10 seconds, so 5-10 seconds still sounds like a very long time to me, but I haven't made time to find another solution. On the other hand, we don't want to add to the existing CI flakiness as you mentioned. If this test generally isn't taking a long time to run it's probably fine; let's just keep an eye on that.